### PR TITLE
banyan-macos-v3

### PR DESCRIPTION
**Pull request by fjamal15**
Discussed with Tanner. The following updates have been made:
- Added a stop function to stop the app before upgrades
- Updated for PKG, not DMG
- Removed ARM logic

### CODE REVIEW

**Comment by fjamal15** [comment]
Also removed LaunchAgent function taht was not being called

File: banyan-macos-v3 [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	killall Banyan
+	sleep 2
+}
+
+
+if [[ "$INVITE_CODE" = "upgrade" && "$DEPLOYMENT_KEY" = "upgrade" ]]; then
+	echo "Running upgrade flow"
+	stop_app
+	download_install
+	start_app
```
Comment (themethod) [inline banyan-macos-v3:119]: `start_app` can be moved outside of the `if` statement since both conditions call it.

File: banyan-macos-v3 [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+function stop_app() {
+	echo "Stopping Banyan app"
+	killall Banyan
+	sleep 2
+}
+
+
+if [[ "$INVITE_CODE" = "upgrade" && "$DEPLOYMENT_KEY" = "upgrade" ]]; then
+	echo "Running upgrade flow"
+	stop_app
```
Comment (themethod) [inline banyan-macos-v3:117]: `stop_app` can be moved before the `if` statement.

File: banyan-macos-v3 [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+	download_install
+	start_app
+else
+	echo "Running zero-touch install flow"
+        stop_app
```
Comment (themethod) [inline banyan-macos-v3:122]: remove

File: banyan-macos-v3 [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        create_config
+	download_install
+	stage
+	start_app
```
Comment (themethod) [inline banyan-macos-v3:126]: remove

File: banyan-macos-v3 [hunk 5]
```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
+
```
Comment (themethod) [inline banyan-macos-v3:2]: Add: `set -euo pipefail`

bash 'strict' mode. Makes the script exit if there is an error.

File: banyan-macos-v3 [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+	sed -i '' "s/REPLACE_WITH_INVITE_CODE/${INVITE_CODE}/" "${global_config_file}"
+	sed -i '' "s/REPLACE_WITH_USER/${deploy_user}/" "${global_config_file}"
+	sed -i '' "s/REPLACE_WITH_EMAIL/${deploy_email}/" "${global_config_file}"
+}
+
+
+function download_install() {
+	echo "Downloading installer PKG"
+
+	full_version="${APP_VERSION}"
```
Comment (themethod) [inline banyan-macos-v3:79]: declare local variables in functions with `local` keyword

**Review by themethod — CHANGES_REQUESTED** [review]

**Comment by fjamal15** [comment]
Closing PR as we are merging into the other script. Have resolved most comments

**Pull request closed without merging**